### ISSUE TITLE
Question on attestation..

### ISSUE THREAD

**Issue by sf-troot** [go-piv/go-ykpiv#30]
The method VerifyAttestation takes two certs as input but I don't see where 'attestedCert' is used, should it be here?:
https://github.com/go-piv/go-ykpiv/blob/32315ce599d8e21e91880e6ecdce55a98420ddc0/attestation.go#L83
or here?:
https://github.com/go-piv/go-ykpiv/blob/32315ce599d8e21e91880e6ecdce55a98420ddc0/attestation.go#L85

**Comment by jtakkala** [comment]
Good catch, looks like `attestedCert` should be on line 85.

**Comment by jtakkala** [comment]
This mistake basically means that VerifyAttestation() would always return true provided any valid attestation cert is passed in, even if the attested cert is not valid. Would be nice to add tests to any fix to prevent a regression here in the future.

**Comment by sf-troot** [comment]
Another question:  
https://github.com/go-piv/go-ykpiv/blob/32315ce599d8e21e91880e6ecdce55a98420ddc0/attestation.go#L81
In this line the attestationCert is marked as CA and added as an Intermediate, would this stop the 'chain' validation since the CARoot is also added as a root?  I am thinking for a proper chain validation this line should not be here?  or am I miss understanding something?

**Comment by jtakkala** [comment]
> In this line the attestationCert is marked as CA and added as an Intermediate, would this stop the 'chain' validation since the CARoot is also added as a root? I am thinking for a proper chain validation this line should not be here? or am I miss understanding something?

No, the reason it’s explicitly set here is because some (all?) Yubikey’s don’t set the isCA attribute on the attestation cert, so chain verification would fail. I believe all intermediate CA certs need the isCA attribute set to true.
